components/layout/display_list_builder.rs

```diff
@@ -567,6 +573,148 @@ fn build_border_radius_for_inner_rect(outer_rect: &Rect<Au>,
+    //
+    // FIXME(#3908, pcwalton): Make sure later stops can't be behind earlier stops.
+    let mut stops = Vec::with_capacity(gradient_stops.len());
+    let mut stop_run = None;
+    for (i, stop) in gradient_stops.iter().enumerate() {
+        let offset = match stop.position {
+            None => {
+                if stop_run.is_none() {
+                    // Initialize a new stop run.
+                    let start_offset = if i == 0 {
+                        0.0
+                    } else {
+                        // `unwrap()` here should never fail because this is the beginning of
+                        // a stop run, which is always bounded by a length or percentage.
+                        position_to_offset(gradient_stops[i - 1].position.unwrap(), length)
+                    };
+                    let (end_index, end_offset) =
+                        match gradient_stops[i..]
+                                        .iter()
+                                        .enumerate()
+                                        .find(|&(_, ref stop)| stop.position.is_some()) {
+                            None => (gradient_stops.len() - 1, 1.0),
+                            Some((end_index, end_stop)) => {
+                                // `unwrap()` here should never fail because this is the end of
+                                // a stop run, which is always bounded by a length or
+                                // percentage.
+                                (end_index,
+                                    position_to_offset(end_stop.position.unwrap(), length))
+                            }
+                        };
+                    stop_run = Some(StopRun {
+                        start_offset: start_offset,
+                        end_offset: end_offset,
+                        start_index: i,
+                        stop_count: end_index - i,
+                    })
+                }
+
+                let stop_run = stop_run.unwrap();
+                let stop_run_length = stop_run.end_offset - stop_run.start_offset;
+                if stop_run.stop_count == 0 {
+                    stop_run.end_offset
+                } else {
+                    stop_run.start_offset +
+                        stop_run_length * (i - stop_run.start_index) as f32 /
+                            (stop_run.stop_count as f32)
+                }
+            }
+            Some(position) => {
+                stop_run = None;
+                position_to_offset(position, length)
+            }
+        };
+        stops.push(GradientStop {
+            offset: offset,
+            color: style.resolve_color(stop.color).to_gfx_color()
+        })
+    }
+    stops
+}
+
+/// Returns the the distance to the nearest or farthest corner depending on the comperator.
+fn get_distance_to_corner<F>(size: &Size2D<Au>, center: &Point2D<Au>, cmp: F) -> Au
+    where F: Fn(f32, f32) -> f32 {
```

Comment:
nit: either brace or `where` to the first column in the next line, here and below.